_specifications/lsp/3.18/types/textEdit.md

```diff
@@ -79,3 +81,31 @@ export interface AnnotatedTextEdit extends TextEdit {
 	annotationId: ChangeAnnotationIdentifier;
 }
 ```
+
+Since 3.18.0, there is also the concept of a snippet text edit, which supports inserting a snippet instead of plain text.
+
+<div class="anchorHolder"><a href="#snippetTextEdit" name="snippetTextEdit" class="linkableAnchor"></a></div>
+
+```typescript
+/**
+ * An interactive text edit.
+ *
+ * @since 3.18.0
+ */
+export interface SnippetTextEdit {
+	/**
+	 * The range of the text document to be manipulated.
+	 */
+	range: Range;
+
+	/**
+	 * The snippet to be inserted.
+	 */
+	snippet: StringValue;
```

Comment:
@dbaeumer It seems like `StringValue`s haven't been documented yet, despite already being used in the metamodel for `InlineValue`s. Should we add that specification in this PR as well?

Reply:
Good point. I would add it to this PR.

Reply:
I decided to create a separate PR: https://github.com/microsoft/language-server-protocol/pull/1893 ;)